src/components/Timer.vue

```diff
@@ -212,44 +221,61 @@ class Timer extends Vue {
     }
 
     @Watch('detailsShown', { immediate: true })
-    @Watch('alwaysShowTime', { immediate: true })
+    @Watch('alwaysShowTime')
     private _setRadius() {
         this.radius.tweenTo(this.detailsShown || this.alwaysShowTime
             ? Timer.RADIUS_GROWTH_FACTOR * Timer.BASE_RADIUS
             : Timer.BASE_RADIUS, 300);
-        this._rerender();
+        this._animateRadius();
     }
 
     @Watch('startTime', { immediate: true })
-    @Watch('endTime', { immediate: true })
-    @Watch('timeOffset', { immediate: true })
+    @Watch('endTime')
+    @Watch('timeOffset')
     private _setTimer() {
         this.sampledTime = Date.now() + this.timeOffset;
-        clearTimeout(this.timeout);
+        clearTimeout(this.timeoutId);
         if (this.startTime && this.endTime) {
-            this.timeout = window.setTimeout(() => this.$emit(Timer.Events.END, this.endTime),
+            this.timeoutId = window.setTimeout(() => this.$emit(Timer.Events.END, this.endTime),
                 this.endTime - this.sampledTime);
+            this._setupUpdateInterval();
         }
-        this._rerender();
     }
 
-    private _rerender() {
-        if (this.requestAnimationFrameId !== null) return;
-        this.requestAnimationFrameId = requestAnimationFrame(() => {
-            const sampledTime = Date.now() + this.timeOffset;
+    @Watch('_updateInterval')
+    private _setupUpdateInterval() {
+        if (this._timeLeft === 0 || this.requestAnimationFrameId !== null) return;
 
-            // update values if necessary
-            if (!this.sampledTime || sampledTime - this.sampledTime >= this._updateInterval
-                || !this.radius.finished) { // animating radius
-                this.sampledTime = sampledTime;
-                this.fullCircleLength = 2 * Math.PI * this.radius.currentValue;
-            }
+        clearInterval(this.updateIntervalId);
+        this.updateIntervalId = window.setInterval(() => {
+            this._rerender();
+            if (this._timeLeft !== 0) return;
+            clearInterval(this.updateIntervalId);
```

Comment:
```suggestion
            if (this._timeLeft === 0) clearInterval(this.updateIntervalId);
```

Reply:
We usually only have `return`s without opening a block.

Reply:
Well, you can put a block around it :stuck_out_tongue: I wanted to change it to not return early, but instead make the `clearInterval` the result of the condition. I was first confused why you are clearing the interval _inside_ the interval, and only later saw that you return early in the regular case. I think early returns should be reserved for exceptions, errors, not the expected flow.